### PR TITLE
Materials core API test 

### PR DESCRIPTION
**Pull request by munrojm**
This PR updates the API with an endpoint for data in the `materials.core` collection. It is intended to enable initial testing.

### CODE REVIEW

File: src/mp_api/materials/query_operators.py [hunk 1]
```diff
@@ -34,3 +35,113 @@ def query(
...
+
+        if deprecated:
+            crit.update({"deprecated": deprecated})
+
+        return {"criteria": crit}
+
+
+class MinMaxQuery(QueryOperator):
+    """
+    Method to generate a query for quanties with a definable min and max
```
Comment (tschaume) [inline src/mp_api/materials/query_operators.py:64]: "quantities"

File: src/mp_api/materials/query_operators.py [hunk 2]
```diff
@@ -34,3 +35,113 @@ def query(
...
+    """
+
+    def query(
+        self,
+        deprecated: Optional[bool] = Query(
+            None, description="Whether the material is marked as deprecated",
+        ),
+    ) -> STORE_PARAMS:
+        """
+        Pagination parameters for the API Endpoint
```
Comment (tschaume) [inline src/mp_api/materials/query_operators.py:52]: This docstring seems to be the same for all `query` functions?

File: src/mp_api/materials/query_operators.py [hunk 3]
```diff
@@ -34,3 +35,113 @@ def query(
...
+
+        d = {
+            "nsites": [nsites_min, nsites_max],
+            "volume": [volume_min, volume_max],
+            "density": [density_min, density_max],
+        }
+
+        for entry in d:
+            if d[entry][0]:
+                if entry not in crit:
```
Comment (tschaume) [inline src/mp_api/materials/query_operators.py:102]: This `if-else` could be saved if `crit` was a `defaultdict(dict)`

File: src/mp_api/tasks/models.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    EMPTY = ("",)
+    GGA_NSCF_Line = ("GGA NSCF Line",)
+    GGA_NSCF_Uniform = ("GGA NSCF Uniform",)
+    GGA_Static = ("GGA Static",)
+    GGA_Structure_Optimization = ("GGA Structure Optimization",)
+    GGA_U_NSCF_Line = ("GGA+U NSCF Line",)
+    GGA_U_NSCF_Uniform = ("GGA+U NSCF Uniform",)
+    GGA_U_Static = ("GGA+U Static",)
+    GGA_U_Structure_Optimization = "GGA+U Structure Optimization"
```
Comment (tschaume) [inline src/mp_api/tasks/models.py:25]: Why is the enum value not a `tuple` here?

File: src/mp_api/tasks/models.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+
+class outputDoc(BaseModel):
+    structure: Structure = Field(
+        None,
+        title="Output Structure",
+        description="Output Structure from the VASP calculation",
+    )
+
+    density: float = Field(..., description="Density of in unites of g/cc")
```
Comment (tschaume) [inline src/mp_api/tasks/models.py:64]: "units" (same typos below)

File: src/mp_api/tasks/models.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    )
+
+    orig_inputs: OrigInputs = Field(
+        None,
+        description="The exact set of input parameters used to generate the current task document.",
+    )
+
+    output: outputDoc = Field(
+        None,
+        description="The exact set of input parameters used to generate the current task document.",
```
Comment (tschaume) [inline src/mp_api/tasks/models.py:98]: "output parameters"

**Comment by munrojm** [comment]
Thanks @tschaume! @jmmshn, are the changes suggested for the the code associated with the `tasks` endpoint okay with you? If so, I'll go ahead and make the changes.

File: src/mp_api/tasks/models.py (hunk 4 above)
Comment (jmmshn) [reply src/mp_api/tasks/models.py:25]: Sorry there's no reason for this. Also for some reason, it seems to do the same thing whether it's a tuple or not???

Comment (jmmshn) [reply src/mp_api/tasks/models.py:25]: @munrojm  please add brackets around the last line here. Sorry for the oversight.

Comment (tschaume) [reply src/mp_api/tasks/models.py:25]: The member value does not need to be a tuple. It's probably fine here to just be the [descriptive string](https://docs.python.org/3/library/enum.html#using-a-descriptive-string).

Comment (jmmshn) [reply src/mp_api/tasks/models.py:25]: OK I figured out what happened.  didn't have these as tuples but I have a comma after each line except the last.  Then the automatic linting added the brackets for everyone except the last line.
But these got parsed by fastAPI the same way (for some reason I do not understand) so I never caught the change.

**Pull request merged**